### ISSUE TITLE
Animated Tiles from Tiled causing error [tilemap.py]

### ISSUE THREAD

**Issue by humansHS** [pvcraven/arcade#603]
## Bug Report

Loading animated tiles causes an error and crash.

### Actual behavior:

`AttributeError: 'NoneType' object has no attribute 'source'`

### Steps to reproduce/example code:

For myself, any time I try to use animated tiles (I am using a tilesheet and not individual tiles, not sure if that should matter), using the following code:

```
  File "C:\Users\John\Desktop\TestGame\testing.py", line 100, in setup
    self.background_list =  arcade.tilemap.process_layer(my_map, background_layer_name, TILE_SCALING)

  File "C:\ProgramData\Anaconda3\lib\site-packages\arcade\tilemap.py", line 393, in process_layer
    return _process_tile_layer(map_object, layer, scaling, base_directory)

  File "C:\ProgramData\Anaconda3\lib\site-packages\arcade\tilemap.py", line 352, in _process_tile_layer
    base_directory=base_directory)

  File "C:\ProgramData\Anaconda3\lib\site-packages\arcade\tilemap.py", line 149, in _create_sprite_from_tile
    tmx_file = tile.image.source

AttributeError: 'NoneType' object has no attribute 'source'
```

The relevant section of code where it errors:

```
def _create_sprite_from_tile(map_object, tile: pytiled_parser.objects.Tile,
                             scaling,
                             base_directory: str = None):

    if base_directory:
        tmx_file = base_directory + tile.image.source
    else:
        tmx_file = tile.image.source
```
the "tile" in question comes from _get_tile_from_gid()

```
            tile = _get_tile_by_gid(map_object, item)
            if tile is None:
                print(f"Warning, couldn't find tile for item {item} in layer "
                      f"'{layer.name}' in file '{map_object.tmx_file}'.")
                continue

            my_sprite = _create_sprite_from_tile(map_object, tile, scaling=scaling,
                                                 base_directory=base_directory)
```

I can't say I understand enough about what is happening here (tilemap.py at around line 100), but I believe the problem is here. Hopefully someone who knows tilemap.py can have a look?

```
    for tileset_key, tileset in map_object.tile_sets.items():

        if tile_gid < tileset_key:
            continue

        tile_ref = tileset.tiles.get(tile_gid - tileset_key)

        # No specific tile info, but there is a tile sheet
        if tile_ref is None \
                and tileset.image is not None \
                and tileset_key <= tile_gid < tileset_key + tileset.tile_count:
            image = pytiled_parser.objects.Image(source=tileset.image.source)
            tile_ref = pytiled_parser.objects.Tile(id_=(tile_gid - tileset_key),
                                                   image=image)

        if tile_ref is None:
            continue
        my_tile = copy.copy(tile_ref)
        my_tile.tileset = tileset
        my_tile.flipped_vertically = flipped_vertically
        my_tile.flipped_diagonally = flipped_diagonally
        my_tile.flipped_horizontally = flipped_horizontally
        return my_tile
    return None
```

**Comment by pvcraven** [comment]
Any chance you can upload a zipped up small example?

**Comment by humansHS** [comment]
Oh is it not erroring for you? Here is an example. If you open the map in tiled and remove the animated 'coins' (purple blobs) then it runs with no crashes, if you edit the tilset and remove the animation it works fine too. Only has the crash for animated tiles.

[test.zip](https://github.com/pvcraven/arcade/files/4254359/test.zip)

**Comment by pvcraven** [comment]
I think it is the combination of using a tilesheet and animated tiles. I've only tested animations using individual images rather than tilesheets. Thanks for the zip file, this will help a lot.

**Comment by pvcraven** [comment]
Should be fixed in 2.3.7. Please test.

**Comment by humansHS** [comment]
Great, it is working perfectly! Thanks so much! Can't wait to easily add animated tiles to my game now! Spinning fans and flashing lights makes it so much better :D

It did take me a little to realize I was missing a key implementation step, didn't realize you had to call "update_animation()" rather than have it in "draw()" or "update()" ... I assume this is to save performance for unanimated tiles?
